siliconcompiler/schema/schema_cfg.py

```diff
@@ -763,42 +764,465 @@ def schema_pdk(cfg, stackup='default'):
+                "cli: -io_standard 'mif0 ddr'",
+                "api: chip.set('datasheet', 'io', 'mif0', 'standard', 'ddr')"],
+            schelp="""Datasheet: IO standard""")
+
+    scparam(cfg, ['datasheet', 'io', name, 'gen'],
+            sctype='[str]',
+            shorthelp="Datasheet: io generation",
+            switch="-io_gen 'name <str>'",
+            example=[
+                "cli: -io_gen 'ddr 3'",
+                "api: chip.set('datasheet', 'io', 'ddr', 'gen', '3')"],
+            schelp="""Datasheet: list of IO standard generations supported.""")
+
+    metrics = {'speed': ['speed', 100, 'MHz'],
+               'width': ['width', 4, None],
+               'channels': ['channels', 4, None]
+               }
+
+    for i, v in metrics.items():
+        scparam(cfg, ['datasheet', 'io', name, i],
+                unit=v[2],
+                sctype='float',
+                shorthelp=f"Datasheet: io {v[0]}",
+                switch=f"-io_{i} 'name <int>'",
+                example=[
+                    f"cli: -io_{i} 'name {v[1]}'",
+                    f"api: chip.set('datasheet', 'io', name, '{i}', {v[1]})"],
+                schelp=f"""IO {v[1]}.
+                """)
+
+    ######################
+    # Processor
+    ######################
+
+    scparam(cfg, ['datasheet', 'proc', name, 'arch'],
+            sctype='str',
+            shorthelp="Datasheet: processor architecture",
+            switch="-proc_arch '0 <str>'",
+            example=[
+                "cli: -proc_arch '0 RV64GC'",
+                "api: chip.set('datasheet', 'proc', name, 'arch', 'openfpga')"],
+            schelp="""Processor architecture.
+            """)
+
+    scparam(cfg, ['datasheet', 'proc', name, 'features'],
+            sctype='[str]',
+            shorthelp="Datasheet: processor features",
+            switch="-proc_features 'name <str>'",
+            example=[
+                "cli: -proc_features '0 SIMD'",
+                "api: chip.set('datasheet','proc','cpu','features', 'SIMD')"],
+            schelp="""List of maker specified processor features.""")
+
+    scparam(cfg, ['datasheet', 'proc', name, 'datatypes'],
+            sctype='[enum]',
+            enum=['int1', 'int4', 'int8', 'int16', 'int32', 'int64', 'int128',
+                  'sint4', 'sint8', 'sint16', 'sint32', 'sint64', 'sint128',
+                  'bfloat16', 'fp16', 'fp32', 'fp64'],
+            shorthelp="Datasheet: processor datatypes",
+            switch="-proc_datatypes 'name <str>'",
+            example=[
+                "cli: -proc_datatypes '0 SIMD'",
+                "api: chip.set('datasheet','proc','cpu','features', 'SIMD')"],
+            schelp="""List of maker specified processor features.""")
+
+    metrics = {'archsize': ['architecture size', 64, None],
+               'cores': ['number of cores', 4, None],
+               'speed': ['maximum frequency', 100, 'MHz'],
+               'ops': ['operations per cycle per core', 4, None],
```

Comment:
are these not overlapping with other keys in the datasheet?

Reply:
somewhat. the 'proc' number is specified on a per block/core basis. For example, there can be a big and little core  runnning at different frequencies. The top level number is the "marketing" number for the device.